src/NuGet.Core/NuGet.Configuration/PackageSource/PackageSourceProvider.cs

```diff
@@ -667,7 +668,16 @@ public void SavePackageSources(IEnumerable<PackageSource> sources, PackageSource
 
             var disabledSourcesSection = Settings.GetSection(ConfigurationConstants.DisabledPackageSources);
             var existingDisabledSources = disabledSourcesSection?.Items.OfType<AddItem>();
-            var existingDisabledSourcesLookup = existingDisabledSources?.ToDictionary(setting => setting.Key, StringComparer.OrdinalIgnoreCase);
+            Dictionary<string, AddItem> existingDisabledSourcesLookup = null;
+
+            try
+            {
+                existingDisabledSourcesLookup = existingDisabledSources?.ToDictionary(setting => setting.Key, StringComparer.OrdinalIgnoreCase);
+            }
+            catch (ArgumentException e)
```

Comment:
Could you use more specific exception instead of generic ArgumentException?
I believe I saw once exception for duplicate entries in dictionary.

Reply:
https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.idictionary-2?view=net-7.0 ArgumentException is the exception thrown with duplicate key in a dictionary